cloud/services/routetables/routetables.go

```diff
@@ -33,40 +34,51 @@ func (s *Service) Reconcile(ctx context.Context) error {
 	}
 
 	for _, rtSpec := range s.Scope.RouteTableSpecs() {
-		existingRouteTable, err := s.Get(ctx, s.Scope.ResourceGroup(), rtSpec.Name)
-		if !azure.ResourceNotFound(err) {
+		if rtSpec.Role == azure.RouteTableControlPlane {
```

Comment:
instead of doing this logic here, can we just have the subnet be part of the route table spec directly?

Reply:
I mean instead of Role, have Subnet in https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/816/files#diff-e0a02b9b279470c6d5411301fba575a9R77

That way no need to have logic that knows about roles here

Reply:
got it, makes sense, will change that

Reply:
made the change, please take a look

Reply:
looks good